syntax = "proto3";
package rl_algorithm_prototype;
import "proto/types.proto";

message InitRequest {
  uint32 id = 1;
  repeated string data = 2;
}

message InitResponse {
  ResultType result = 1;
}

message ActRequest {
  Observation observation = 1;
}

message ActResponse {
  Action action = 1;
}

message ValueFuncUpdateRequest {
  ValueFuncUpdate value_func_update = 1;
}

message ValueFuncUpdateResponse {
  ResultType result = 1;
}

message PolicyUpdateRequest {
  PolicyUpdate policy_update = 1;
}

message PolicyUpdateResponse {
  ResultType result = 1;
}

message GetValueFuncRequest {
  string data = 1;
}

message GetValueFuncResponse {
  ValueFuncion value_function = 1;
}

message GetPolicyRequest {
  string data = 1;
}

message GetPolicyResponse {
  Policy policy = 1;
}